src/main/java/org/datadog/jenkins/plugins/datadog/clients/DogStatsDClient.java

```diff
@@ -118,6 +104,22 @@ private DogStatsDClient(String hostname, Integer port, Integer logCollectionPort
         reinitializeLogger(true);
     }
 
+    public static void validateCongiguration(String hostname, Integer port, Integer logCollectionPort) throws RuntimeException {
+        if (hostname == null || hostname.isEmpty()) {
+            throw new RuntimeException("Datadog Target URL is not set properly");
+        }
+        if (!DatadogUtilities.isValidHostname(hostname)) {
+            logger.severe("Invalid Hostname. Your hostname is invalid, likely because it violates the format set in RFC 1123.");
```

Comment:
Why no exception here?

Reply:
I'm pretty sure this was previous behavior, but I can look into it more.

Reply:
I looked into and I think this is correct. I updated it to not create an instance if any config is broken.